src/commands/publish.ts

```diff
@@ -82,103 +93,140 @@ async function checkREADME(config: Config) {
+    options.entry && (cfg.entry = options.entry);
+    options.unstable && (cfg.unstable = options.unstable);
+    options.unlisted && (cfg.unstable = options.unlisted);
+    options.repository &&
+      (cfg.repository = urlType("repository", {}, options.repository));
+    options.files && (cfg.files = options.files);
+    options.ignore && (cfg.ignore = parseIgnore(options.ignore.join()));
+    options.checkFmt && (cfg.files = options.files);
+    options.checkTests && (cfg.checkTests = options.checkTests);
+    options.checkInstallation &&
+      (cfg.checkInstallation = options.checkInstallation);
+    options.checkAll && (cfg.checkAll = options.checkAll);
+    return cfg;
+  } catch (err) {
+    log.error(err);
+    return;
   }
-
-  return matched.map((el) =>
-    [el, readFileBtoa(el.fullPath)] as [typeof el, string]
-  ).reduce((p, c) => {
-    p[c[0].path] = c[1];
-    return p;
-  }, {} as { [x: string]: string });
 }
 
-function checkEntry(config: Config, matched: File[]) {
-  if (config.entry) {
-    config.entry = config.entry?.replace(/^[.]/, "").replace(
-      /^[^/]/,
-      (s: string) => `/${s}`,
+async function checkUp(
+  config: Config,
+  matched: MatchedFile[],
+): Promise<boolean> {
+  if (config.checkFmt || config.fmt || config.checkAll) {
+    const process = Deno.run(
+      { cmd: ["deno", "fmt"], stderr: "null", stdout: "null" },
     );
+    const status = await process.status();
+    if (status.success) {
+      log.info("Formatted your code.");
+    } else {
+      log.error(`${italic("deno fmt")} returned a non-zero code.`);
+      return false;
+    }
   }
-  if (!matched.find((e) => e.path === config.entry || "/mod.ts")) {
-    log.error(
-      `No ${config.entry || "/mod.ts"} found. This file is required.`,
+
+  if (config.checkTests || config.checkAll) {
+    const process = Deno.run(
+      {
+        cmd: ["deno", "test", "-A", "--unstable"],
+        stderr: "null",
+        stdout: "piped",
+      },
     );
-    return true;
+    const status = await process.status();
+    const stdout = new TextDecoder("utf-8").decode(await process.output());
+    if (status.success) {
+      log.info("Tests passed successfully.");
+    } else {
+      if (stdout.match(/^No matching test modules found/)) {
+        log.info("No matching test modules found, tests skipped.");
+      } else {
+        log.error(`${italic("deno test")} returned a non-zero code.`);
+        return false;
+      }
+    }
   }
-}
 
-function isVersionStable(v: string) {
-  return !!((semver.major(v) === 0) || semver.prerelease(v));
+  if (config.checkInstallation || config.checkAll) {
+    const tempDir = await Deno.makeTempDir();
+    for (let i = 0; i < matched.length; i++) {
+      const file = matched[i];
+      const dir = join(tempDir, dirname(file.path));
+      try {
+        await Deno.mkdir(dir, { recursive: true });
+      } catch (err) {
+        if (err !== Deno.errors.AlreadyExists) {
```

Comment:
`Deno.errors.AlreadyExists` is a class - should use instanceof
```suggestion
        if (!(err instanceof Deno.errors.AlreadyExists)) {
```

Reply:
Yes, of course, you're right
I'm not thinking straight, I've been working too long on this today 😄